src/status_im/contexts/communities/actions/invite_contacts/view.cljs

```diff
@@ -0,0 +1,131 @@
+(ns status-im.contexts.communities.actions.invite-contacts.view
+  (:require
+    [quo.core :as quo]
+    [quo.foundations.resources :as resources]
+    [quo.theme]
+    [react-native.core :as rn]
+    [react-native.gesture :as gesture]
+    [react-native.share :as share]
+    [status-im.common.contact-list-item.view :as contact-list-item]
+    [status-im.common.contact-list.view :as contact-list]
+    [status-im.contexts.communities.actions.invite-contacts.style :as style]
+    [status-im.contexts.communities.events :as communities]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(defn- no-contacts-view
+  [{:keys [theme id]}]
+  (let [customization-color             (rf/sub [:profile/customization-color])
+        {:keys [universal-profile-url]} (rf/sub [:profile/profile])
+        on-press-share-community        (rn/use-callback
+                                         #(rf/dispatch [:communities/share-community-url-with-data id]))
+        on-press-share-profile          (rn/use-callback #(share/open {:url universal-profile-url}))]
+    (fn []
+      [rn/view
+       {:style (style/no-contacts)}
+       [rn/image {:source (resources/get-themed-image :no-contacts-to-chat theme)}]
+       [quo/text
+        {:weight :semi-bold
+         :size   :paragraph-1
+         :style  style/no-contacts-text}
+        (i18n/label :t/you-have-no-contacts)]
+       [quo/text
+        {:weight :regular
+         :size   :paragraph-2}
+        (i18n/label :t/dont-yell-at-me)]
+       [quo/button
+        {:customization-color customization-color
+         :theme               theme
+         :type                :primary
+         :size                32
+         :container-style     style/no-contacts-button-container
+         :on-press            on-press-share-community}
+        (i18n/label :t/send-community-link)]
+       [quo/button
+        {:customization-color customization-color
+         :theme               theme
+         :type                :grey
+         :size                32
+         :on-press            on-press-share-profile}
+        (i18n/label :t/invite-friends-to-status)]])))
+
+(defn- contact-item-render
+  [_]
+  (fn [{:keys [public-key] :as item}]
+    (let [user-selected? (rf/sub [:is-contact-selected? public-key])
+          on-toggle      #(if user-selected?
+                            (rf/dispatch [:deselect-contact public-key])
+                            (rf/dispatch [:select-contact public-key]))]
+      [contact-list-item/contact-list-item
+       {:on-press                on-toggle
+        :allow-multiple-presses? true
+        :accessory               {:type     :checkbox
+                                  :checked? user-selected?
+                                  :on-check on-toggle}}
+       item])))
+
+(defn view-internal
+  [{:keys [theme]}]
+  (let [{:keys [id]}          (rf/sub [:get-screen-params])
+        {:keys [name images]} (rf/sub [:communities/community id])
```

Comment:
What if the community name changes or the images? This component won't re-render. The majority of the time, moving subs outside the renderer should be done only for data that's stable across re-renders. For example, we know the community ID won't ever change, so that's fine. The `name` and `images` props are not stable, hence this will be a bug.

Reply:
Done